### PR TITLE
Rules and questions pages

### PR DESCRIPTION
**Pull request by yaswanthmyneni**
# Rules and Questions pages

### CODE REVIEW

**Review by sri-rishi — APPROVED** [review]
Great work, code looks good , feel free to merge

File: src/components/routing/Routing.jsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import { CategoryPage, HomePage, RulesPage, QuestionPage } from "pages";
 import { Routes, Route } from "react-router-dom";
 
 const Routing = () => {
   return (
     <Routes>
       <Route path="/" element={<HomePage />} />
       <Route path="/categories" element={<CategoryPage />} />
+      <Route path="/rules" element={<RulesPage />} />
+      <Route path="/quiz" element={<QuestionPage />} />
```
Comment (Ankur9669) [inline src/components/routing/Routing.jsx:10]: Later you can wrap the rules page and question page in requiredAuth

File: src/pages/question-page/QuestionPage.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    questionNumber: 0,
+    userAnsweredIndex: null,
+    answer: {},
+  });
+  const { questionNumber, userAnsweredIndex, answer } = state;
+
+  // from react-router-dom
+  const navigate = useNavigate();
+
+  // from quiz context
```
Comment (Ankur9669) [inline src/pages/question-page/QuestionPage.jsx:18]: Appreciation: Nice use of comments

File: src/pages/question-page/QuestionPage.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    quizState: { questions },
+    quizDispatch,
+  } = useQuizContext();
+
+  return (
+    <main className="question-page-main">
+      <div className="flex flex-direction">
+        <h2 className="text-center m-b-24">Quiz Name</h2>
+        {questions.map((questionDetails, questionIndex) => {
+          if (questionIndex === questionNumber) {
```
Comment (Ankur9669) [inline src/pages/question-page/QuestionPage.jsx:29]: Here you can use ternary operator although this is also fine

**Review by Ankur9669 — APPROVED** [review]
The UI and code looks good to me, I have given some suggestions other than that feel free to merge the PR

**Pull request merged**